packages/lib/src/components/MBWay/components/MBWayInput/MBWayInput.tsx

```diff
@@ -1,69 +1,74 @@
 import { h } from 'preact';
-import { useState, useEffect } from 'preact/hooks';
-import classNames from 'classnames';
+import { useState, useLayoutEffect, useRef } from 'preact/hooks';
 import useCoreContext from '../../../../core/Context/useCoreContext';
-import Field from '../../../internal/FormFields/Field';
-import { renderFormField } from '../../../internal/FormFields';
-import { MBWayDataState, MBWayInputProps } from './types';
+import { MBWayInputProps } from './types';
 import './MBWayInput.scss';
-import useForm from '../../../../utils/useForm';
-const phoneNumberRegEx = /^[+]*[0-9]{1,4}[\s/0-9]*$/;
+import getDataset from '../../../../core/Services/get-dataset';
+import { DataSet } from '../../../../core/Services/data-set';
+import PhoneInput from '../../../internal/PhoneInputNew';
+import AdyenCheckoutError from '../../../../core/Errors/AdyenCheckoutError';
+import LoadingWrapper from '../../../internal/LoadingWrapper';
 
 function MBWayInput(props: MBWayInputProps) {
-    const { i18n } = useCoreContext();
+    const { i18n, loadingContext } = useCoreContext();
 
-    const { handleChangeFor, triggerValidation, data, valid, errors, isValid } = useForm<MBWayDataState>({
-        schema: ['telephoneNumber'],
-        defaultData: props.data,
-        rules: {
-            telephoneNumber: {
-                validate: num => phoneNumberRegEx.test(num) && num.length >= 7,
-                errorMessage: 'mobileNumber.invalid',
-                modes: ['blur']
-            }
-        },
-        formatters: {
-            telephoneNumber: num => num.replace(/[^0-9+\s]/g, '')
-        }
-    });
+    const phoneInputRef = useRef(null);
 
-    const [status, setStatus] = useState('ready');
+    const { allowedCountries = [] } = props;
+
+    const [phonePrefixes, setPhonePrefixes] = useState<DataSet>([]);
+    const [status, setStatus] = useState<string>('loading');
 
     this.setStatus = setStatus;
-    this.showValidation = triggerValidation;
+    this.showValidation = phoneInputRef?.current?.triggerValidation;
+
+    useLayoutEffect(() => {
```

Comment:
Minor thing: seems that this whole logic here could be extracted to a custom hook (ex: `usePhonePrefixes`):

```javascript
const { isLoadingPrefixes, phonePrefixes } = usePhonePrefixes();
```

That can declutter a bit the code :-) It will look simpler and can improve reusability

Reply:
What are you talking about extracting? The phone prefix loading? I'm curious as to why the `this.showValidation` is also highlighted?

Reply:
Sorry I wasn't clear - I didn't intend to highlight the `this.showValidation` part 😅 

I meant the whole logic inside the `useLayouEffect`: fetching the data from the API and handling the loading status of this action. This could be extracted to a custom hook.

Reply:
Good idea - have created a custom hook